solidity/random-beacon/contracts/DKGValidator.sol

```diff
@@ -0,0 +1,230 @@
+    /// @param startBlock DKG start block
+    /// @return isValid true if the result is valid, false otherwise
+    /// @return errorMsg validation error message; empty for a valid result
+    function validate(
+        DKG.Result calldata result,
+        uint256 seed,
+        uint256 startBlock
+    ) external view returns (bool isValid, string memory errorMsg) {
+        (bool hasValidFields, string memory error) = validateFields(result);
+        if (!hasValidFields) {
+            return (false, error);
+        }
+
+        if (!validateSignatures(result, startBlock)) {
+            return (false, "Invalid signatures");
+        }
+
+        if (!validateGroupMembers(result, seed)) {
+            return (false, "Invalid group members");
+        }
+
+        return (true, "");
+    }
+
+    /// @notice Performs a static validation of DKG result fields: lengths,
+    ///         ranges, and order of arrays.
+    /// @return isValid true if the result is valid, false otherwise
+    /// @return errorMsg validation error message; empty for a valid result
+    function validateFields(DKG.Result calldata result)
+        public
+        view
+        returns (bool isValid, string memory errorMsg)
+    {
+        // Group public key needs to be 128 bytes long.
+        if (result.groupPubKey.length != 128) {
+            return (false, "Malformed group public key");
+        }
+
+        // The number of misbehaved members can not exceed the threshold.
+        // Misbehaved member indices needs to be unique, between [1,64],
+        // and sorted in ascending order.
+        uint8[] calldata misbehavedMembersIndices = result
+            .misbehavedMembersIndices;
+        if (groupSize - misbehavedMembersIndices.length < activeThreshold) {
+            return (false, "Too many members misbehaving during DKG");
+        }
+        if (misbehavedMembersIndices.length > 1) {
+            if (
+                misbehavedMembersIndices[0] < 1 ||
+                misbehavedMembersIndices[misbehavedMembersIndices.length - 1] >
+                64
+            ) {
+                return (false, "Corrupted misbehaved members indices");
+            }
+            for (uint256 i = 1; i < misbehavedMembersIndices.length; i++) {
+                if (
+                    misbehavedMembersIndices[i - 1] >=
+                    misbehavedMembersIndices[i]
+                ) {
+                    return (false, "Corrupted misbehaved members indices");
+                }
+            }
+            uint8 last = misbehavedMembersIndices[
+                misbehavedMembersIndices.length - 1
+            ];
+            if (last < 1 || last > groupSize) {
+                return (false, "Corrupted misbehaved members indices");
+            }
+        }
+
+        // Each signature needs to be 65 bytes long and signatures need to be
+        // provided.
+        uint256 signaturesCount = result.signatures.length / 65;
```

Comment:
Maybe we can extract `65` to a constant and name it, for example, `signatureByteSize`?

Reply:
https://github.com/keep-network/keep-core/pull/2734/commits/5bc95364647baabb9b6c7951805b9fb2681dcdfb